yb-voyager/src/errs/targetdb.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errs
+
+import (
+	"fmt"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils/sqlname"
+)
+
+const (
+	// steps
+	IMPORT_BATCH_ERROR_STEP_COPY                         = "copy"
+	IMPORT_BATCH_ERROR_STEP_METADATA_ENTRY               = "metadata_entry"
+	IMPORT_BATCH_ERROR_STEP_CHECK_BATCH_ALREADY_IMPORTED = "check_batch_already_imported"
+	IMPORT_BATCH_ERROR_STEP_OPEN_BATCH                   = "open_batch"
+	IMPORT_BATCH_ERROR_STEP_READ_LINE_BATCH              = "read_line_batch"
+	IMPORT_BATCH_ERROR_STEP_COMMIT_TXN                   = "commit_txn"
+	IMPORT_BATCH_ERROR_STEP_ROLLBACK_TXN                 = "rollback_txn"
+
+	// flows
+	IMPORT_BATCH_ERROR_FLOW_COPY_NORMAL  = "copy_normal"
+	IMPORT_BATCH_ERROR_FLOW_COPY_FAST    = "copy_fast"
+	IMPORT_BATCH_ERROR_FLOW_COPY_RECOVER = "copy_recover"
+)
+
+type ImportBatchError struct {
+	tableName         sqlname.NameTuple
+	batchFilePath     string
+	err               error
+	flow              string
+	step              string
+	dbSpecificContext map[string]string
+}
+
+func (e ImportBatchError) Step() string {
+	return e.step
+}
+
+func (e ImportBatchError) Flow() string {
+	return e.flow
+}
+
+func (e ImportBatchError) Error() string {
+	return fmt.Sprintf("import batch: %q into %s: flow=%s: step=%s: %s %s", e.batchFilePath, e.tableName.ForOutput(), e.flow, e.step, e.err.Error(), utils.MapToString(e.dbSpecificContext))
+}
+
+func (e ImportBatchError) Unwrap() error {
+	return e.err
+}
+
+func NewImportBatchError(tableName sqlname.NameTuple, batchFilePath string, err error, flow, step string, dbSpecificContext map[string]string) ImportBatchError {
+	if dbSpecificContext == nil {
```

Comment:
Instead of getting the map from the caller in some cases, can we directly fetch the DB-specific error from the `err` here itself at a single place, so we know what all we have and what all needs to be sent to callhome?

Reply:
Are you saying that the logic for getting the error-code/etc out of the underlying `pgErr` should live here instead of the caller-side? 

I see your point; one problem was that it's not only pgErr, it could be other types of errors (like for oracle/mysql as well), so I was a little hesitant to keep that logic here. Let me think about this a bit more. Perhaps a helper NewImportBatchErrorForPgErr will help.

Reply:
I think having helpers like :
 - NewImportBatchErrorForPgErr
 - NewImportBatchErrorForOracleErr
 - NewImportBatchErrorForMysqlErr 
each defined in targetdb package (not errs package) will be a good way to do this. 

That way each db's error-handling logic is restricted to it's own package/struct, and the errs struct remains very lightweight. 

@priyanshi-yb what say?

Reply:
will take this up in the followup PR 👍